modules/core/src/lib/mode-handlers/mode-handler.js

```diff
@@ -1,5 +1,12 @@
 // @flow
 
+import turfBearing from '@turf/bearing';
```

Comment:
Looks like you're adding a lot of code here that is specific to translate snapping mode. I wonder if you could create a separate class to encapsulate this functionality. I think the only code that should be in `ModeHandler` is stuff that applies to all (or at least several) modes.

Reply:
@supersonicclay The snapping feature now snaps at polygon vertexes instead of edge midpoints. If polygon snapping is enabled, intermediate edit handles will be rendered to indicate the point that is closest to the selected polygon. Additionally, I abstracted the snapping logic into a new SnapHandler class.

![snap_visuals](https://user-images.githubusercontent.com/42187119/54465205-3507fc80-4737-11e9-9e2a-c8106decce3e.gif)